test/e2e/tests/network/multi-rpc.spec.ts

```diff
@@ -0,0 +1,461 @@
+import { strict as assert } from 'assert';
+import { Suite } from 'mocha';
+import FixtureBuilder from '../../fixture-builder';
+import {
+  defaultGanacheOptions,
+  importSRPOnboardingFlow,
+  regularDelayMs,
+  TEST_SEED_PHRASE,
+  unlockWallet,
+  withFixtures,
+} from '../../helpers';
+import { Driver } from '../../webdriver/driver';
+import { Mockttp } from '../../mock-e2e';
+import {
+  expectMockRequest,
+  expectNoMockRequest,
+} from '../../helpers/mock-server';
+
+describe('MultiRpc:', function (this: Suite) {
+  it('should migrate to multi rpc @no-mmi', async function () {
+    async function mockRPCURLAndChainId(mockServer: Mockttp) {
+      return [
+        await mockServer
+          .forPost('https://responsive-rpc.test/')
+          .thenCallback(() => ({
+            statusCode: 200,
+            json: {
+              id: '1694444405781',
+              jsonrpc: '2.0',
+              result: '0xa4b1',
+            },
+          })),
+      ];
+    }
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder({ onboarding: true })
+          .withNetworkController({
+            providerConfig: {
+              rpcPrefs: { blockExplorerUrl: 'https://etherscan.io/' },
+            },
+            networkConfigurations: {
```

Comment:
It'd be nicer to use `mockNetworkStateOld` here so that we keep the network state going through mock functions.  To avoid having to change it everywhere when the state changes the next time

Reply:
I can try to do that